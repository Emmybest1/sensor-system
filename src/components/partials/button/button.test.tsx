import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { Button } from '..';
import { ReduxProvider } from 'redux/__test__/redux-state.mock';

describe('<Button/>', () => {
  test('should render Component', () => {
    const { container } = render(
      <ReduxProvider internal>
        <Router>
          <Button testId="testBtn" shape="radius" width="80%">
            Radius
          </Button>
          <Button testId="testBtn" shape="block" bg="#333" color="red">
            Block
          </Button>
        </Router>
      </ReduxProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
