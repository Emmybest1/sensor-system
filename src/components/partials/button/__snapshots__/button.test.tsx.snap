// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render Component 1`] = `
<button
  class="sc-gTgzoy gUJGCG"
  color="#fff"
  shape="radius"
  width="80%"
>
  Radius
</button>
`;
